mbuild/formats/lammpsdata.py

```diff
@@ -161,23 +165,37 @@ def write_lammpsdata(structure, filename):
 
         # Atom data
         data.write('\nAtoms\n\n')
-        for i,coords in enumerate(xyz):
-            data.write('{:d}\t{:d}\t{:d}\t{:.6f}\t{:.6f}\t{:.6f}\t{:.6f}\n'.format(i+1,0,unique_types.index(types[i])+1,charges[i],*coords))
-
-        # Bond data
-        if bonds:
-            data.write('\nBonds\n\n')
-            for i,bond in enumerate(bonds):
-                data.write('{:d}\t{:d}\t{:d}\t{:d}\n'.format(i+1,bond_types[i],bond[0],bond[1]))
-
-        # Angle data
-        if angles:
-            data.write('\nAngles\n\n')
-            for i,angle in enumerate(angles):
-                data.write('{:d}\t{:d}\t{:d}\t{:d}\t{:d}\n'.format(i+1,angle_types[i],angle[0],angle[1],angle[2]))
-
-        # Dihedral data
-        if dihedrals:
-            data.write('\nDihedrals\n\n')
-            for i,dihedral in enumerate(dihedrals):
-                data.write('{:d}\t{:d}\t{:d}\t{:d}\t{:d}\t{:d}\n'.format(i+1,dihedral_types[i],dihedral[0],dihedral[1],dihedral[2],dihedral[3]))
+        if atom_style == 'atomic':
+            for i,coords in enumerate(xyz):
+                data.write('{:d}\t{:d}\t{:.6f}\t{:.6f}\t{:.6f}\n'.format(i+1,unique_types.index(types[i])+1,*coords))
+
+        elif atom_style == 'charge':
+            for i,coords in enumerate(xyz):
+                data.write('{:d}\t{:d}\t{:.6f}\t{:.6f}\t{:.6f}\t{:.6f}\n'.format(i+1,unique_types.index(types[i])+1,charges[i],*coords))
+
+        elif atom_style == 'molecular':
+            for i,coords in enumerate(xyz):
+                data.write('{:d}\t{:d}\t{:d}\t{:.6f}\t{:.6f}\t{:.6f}\n'.format(i+1,0,unique_types.index(types[i])+1,*coords))
+
+        elif atom_style == 'full':
+            for i,coords in enumerate(xyz):
+                data.write('{:d}\t{:d}\t{:d}\t{:.6f}\t{:.6f}\t{:.6f}\t{:.6f}\n'.format(i+1,0,unique_types.index(types[i])+1,charges[i],*coords))
```

Comment:
else statement here to catch incorrect atom_styles would be a good addition. Easy to test for too.

Reply:
Another way to reduce some code duplication here:
```python
if atom_style == 'atomic':
    atom_line = '{index:d}\t{type_index:d}\t...{charge:.6f}'
elif ...
...

for i, coords in enumerate(xyz):
    data.write(atom_line.format(index=i+1, type_index=unique_types..., charge=...))
```

When using this keyword style you can always over-specify named arguments, just not under-specify.